ImportStructure.php

```diff
@@ -307,7 +315,9 @@ private function validateAttributes($attributeArray){
         }
         foreach ($attributeArray as $attributeName => $value) {
             if(!in_array($attributeName, $allowedAttributes)) {
-                throw new \Exception("Question attribute '{$attributeName}' is not defined for IMEX and the import breaks here ");
+                // Remove the attribute from the array so that only "safe" attributes remain.
```

Comment:
>  it does nothing else.

It unsets the value. So, instead of stopping, it just ignores non safe attributes, and then continues.

Reply:
Sorry, I have not tested this, so I'm just commenting how this looks. The `$attributeArray` is not returned by the method, also it is not an object property. So what I see is that this unsets attributes, but the array will not reach anywhere further, making it essentially meaningless. No?

Reply:
I think it is passed by reference as being an array, right?

Reply:
No, not passed by reference. Passing by reference is evil :) If the idea is to modify the input array, then the method should unset and then return the filtered array in an obvious and clear way. Also, I would then recommend renaming the method to somth like filterStandardAttributes to address what its actually doing. 

But, I see a problem here that with this solution we will ignore misspelling in standard attributes. If a standard attribute is misspelled, it will be regarded as a custom attribute and it will be imported without any feedback. Which was the initial idea of this validation method - to check that you have not made a mistake in typing the attribute names, and get a big red exception if you have invalid input. 

I am not sure how you handle the custom theme/plugin attributes, but maybe there is a way to inject the custom attributes (via config?) and ensure that the plugin will check that you still have passed what you thought you want to pass and notify you if you have made a typo? I can imagine the headache trying to locate the issue if you think you have imported correct attributes, but it turns out you have mistyped some. It would be really valuable not to pass invalid data here. 

Maybe it's an option to add a plugin configuration eg a json field to manually configure additional allowed attributes instead of just blindly importing anything that is out of standard set